api/src/controllers/reviewController.js

```diff
@@ -0,0 +1,53 @@
+import db from "../db_connection.js";
+import { buildReviewDto } from "../dtos/buildReviewDto.js";
+import { checkReviewInputsValidation } from "../utils/checkReviewInputsValidation.js";
+import { updatePlantAverageRating } from "../utils/updatePlantAverageRating.js";
+
+export const getAllreviewsByPlantId = async (req, res) => {
+  try {
+    const plantId = parseInt(req.params.plantId);
+    if (!plantId || isNaN(plantId) || plantId < 1) {
+      return res.status(400).json({ message: "Invalid plantId" });
+    }
+    const result = await db("reviews").select("*").where({ plant_id: plantId });
+
+    console.log(result);
+
+    if (!result) {
+      return res.status(404).json({ message: "No review found" });
+    }
+
+    const reviews = result.map((item) => buildReviewDto(item));
+
+    res.json(reviews);
+  } catch (err) {
+    console.error("Error fetching plants:", err);

```

Comment:
Error message says "Error fetching plants" but the function fetches reviews. Update to "Error fetching reviews" for clarity.